04-onion-routing.md

```diff
@@ -250,15 +250,15 @@ The reader:
   - if it is not the final node:
     - MUST return an error if:
       - `short_channel_id` is not present,
-       - it cannot forward the HTLC to the peer indicated by the channel `short_channel_id`.
-       - incoming `amount_msat` - `fee` < `amt_to_forward` (where `fee` is the advertised fee as described in [BOLT #7](07-routing-gossip.md#htlc-fees))
-       - `cltv_expiry` - `cltv_expiry_delta` < `outgoing_cltv_value`
+      - it cannot forward the HTLC to the peer indicated by the channel `short_channel_id`.
+      - incoming `amount_msat` - `fee` < `amt_to_forward` (where `fee` is the advertised fee as described in [BOLT #7](07-routing-gossip.md#htlc-fees))
+      - `cltv_expiry` - `cltv_expiry_delta` < `outgoing_cltv_value`
   - if it is the final node:
     - MUST treat `total_msat` as if it were equal to `amt_to_forward` if it
       is not present.
     - MUST return an error if:
-       - incoming `amount_msat` != `amt_to_forward`.
-       - incoming `cltv_expiry` != `cltv_expiry_delta`.
+      - incoming `amount_msat` < `amt_to_forward`.
+      - incoming `cltv_expiry` < `outgoing_cltv_value`.
```

Comment:
Should this be `>` for the final node? For routing nodes a higher expiry is accepted too?

Reply:
Nvm, this are the error conditions.

Reply:
Can you detail? I think the current inequation is correct, you want to return an error if you receive an HTLC that expires before what is specified in the onion, don't you? But you are completely fine with accepting an HTLC that expires after what what the sender intended.

An intermediate node can decide to use a lower `cltv_expiry_delta` than what the sender used when relaying (even honestly, if they recently lowered it), so intermediate nodes could receive an HTLC with a higher `cltv_expiry` than what the sender wanted, but that's a net benefit for them so they shouldn't reject that (and with the current requirements, they won't, because when computing the next `cltv_expiry` they will find something greater than what they're instructed to relay by their `outgoing_cltv_value`).

Does that make sense or should I detail that with some concrete sample values?

Reply:
No it's clear, I thought I was reading the acceptance conditions.